### PR TITLE
Be polite! While on walk intent you won't bump, swap places or push other people now.

### PR DESCRIPTION
**Pull request by GuillaumePrata**
## About The Pull Request
Just the title

## Why It's Good For The Game
We might be used to the jank at this point, but our movement system is kinda atrocious...
You don't have much control on how your character moves around, it feels floaty, people slingshot around the place a lot bumping, swapping or pushing others for 30 tiles down a hallway...

So this is a small change to add some precise control to our characters that is completely optional. The only time we need to walk in game for a mechanical reason, that I can remember, is over wet tiles.

With this change you can be polite and not shove people around interrupting their tasks, I personally will enjoy walking and asking others to walk inside the medbay so they stop pushing doctors around.

Also, thanks Goof for bringing up a point about spessman movement on discord, is what made me stop being lazy for an hour and see how the code works to do this PR.

## Changelog

:cl: Guillaume Prata
qol: Be polite! While on walk intent you won't bump, swap places or push other people now.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -81,6 +81,10 @@
 
 //Called when we bump onto a mob
 /mob/living/proc/MobBump(mob/M)
+	//No bumping/swapping/pushing others if you are on walk intent
+	if(src.m_intent == MOVE_INTENT_WALK)
```
Comment (SuperNovaa41) [inline code/modules/mob/living/living.dm:85]:
```suggestion
	if(m_intent == MOVE_INTENT_WALK)
```
pointless

Comment (GuillaumePrata) [reply code/modules/mob/living/living.dm:85]: Done and thank you.

**Comment by Iamgoofball** [comment]
Make sure that you can't walk past someone with combat mode on, so that combat mode still requires you to rest to get around, putting yourself into a vulnerable state.

**Comment by GuillaumePrata** [comment]
> Make sure that you can't walk past someone with combat mode on, so that combat mode still requires you to rest to get around, putting yourself into a vulnerable state.

I can test it but nothing should change when compared to other players, you just won't try to swap/push people if you are on walk intent regardless of your, or theirs, combat mode.

**Comment by Farquaar** [comment]
It's polite, yes. But is it efficient? Does it coincide with having a plan to kill everyone you meet?

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Please confirm that someone with combat mode on still can't be moved past without resting/being grabbed or wahtever.

**Comment by Palpetinus** [comment]
>The only time we need to walk in game for a mechanical reason, that I can remember, is over wet tiles

Holobarriers

**Comment by GuillaumePrata** [comment]
> Please confirm that someone with combat mode on still can't be moved past without resting/being grabbed or wahtever.

Nothing changes on that end.
And you can still swap/push people that are on walk intent if they are not on combat mode.

**Comment by Mothblocks** [comment]
Okay I will take your word for it

**Pull request merged**